test/infrastructure/docker/internal/docker/kind_manager.go

```diff
@@ -56,15 +54,6 @@ type nodeCreateOpts struct {
 
 // CreateControlPlaneNode will create a new control plane container.
 func (m *Manager) CreateControlPlaneNode(ctx context.Context, name, image, clusterName, listenAddress string, port int32, mounts []v1alpha4.Mount, portMappings []v1alpha4.PortMapping, labels map[string]string, ipFamily clusterv1.ClusterIPFamily) (*types.Node, error) {
```

Comment:
I found this comment for HostPort in the kind API:

```
	// NOTE: if you set the special value of `-1` then the node backend
	// (docker, podman...) will be left to pick the port instead.
	// This is potentially useful for remote hosts, BUT it means when the container
	// is restarted it will be randomized. Leave this unset to allow kind to pick it.
```

So I tried what happens with the variant on this PR vs main.

This PR: After restart containers get a new random port
main: Ports are stable across restarts

As far as I can see we only have host ports for the haproxy and the control plane containers.

Restarting a control plane node breaks connectivity to the apiserver for a few seconds but it recovers. So that's fine.

Restarting haproxy does not break connectivity from mgmt to workload cluster (i.e. the kubeconfig we store still works (endpoint: `https://172.18.0.4:6443`)). It only breaks the kubeconfig from `kind get kubeconfig` (endpoint: `https://0.0.0.0:32776`)

So overall this only breaks kubeconfigs from kind after haproxy restarts.

For me this tradeoff is fine

Reply:
Thanks for finding that - it seems like a reasonable tradeoff, given these clusters aren't production ready and users will be able to fix the kubeconfig if they really need to.

Reply:
Sounds good!